src/DurableTask.Netherite/TransportLayer/EventHubs/EventHubsProcessor.cs

```diff
@@ -47,9 +48,16 @@ class EventHubsProcessor : IEventProcessor, TransportAbstraction.IDurabilityList
         // since EventProcessorHost does not redeliver packets, we need to keep them around until we are sure
         // they are processed durably, so we can redeliver them when recycling/recovering a partition
         // we make this a concurrent queue so we can remove confirmed events concurrently with receiving new ones
-        readonly ConcurrentQueue<(PartitionEvent[] events, long offset, long seqno)> pendingDelivery;
+        readonly ConcurrentQueue<EventEntry> pendingDelivery;
         AsyncLock deliveryLock;
 
+        record struct EventEntry(long SeqNo, long BatchPos, PartitionEvent Event, bool LastInBatch, long Offset, BlockBlobClient BatchBlob)
+        {
+            public EventEntry(long seqNo, long batchPos, PartitionEvent evt) : this(seqNo, batchPos, evt, false, 0, null) { }
+
+            public EventEntry(long seqNo, long batchPos, PartitionEvent evt, long offset, BlockBlobClient blob) : this(seqNo, batchPos, evt, true, 0, blob) { }
```

Comment:
same feedback here about, in the future, specifying the param names for these "inline data" parameter, so we know what they represent :-)